pilot/cmd/pilot-agent/status/server.go

```diff
@@ -216,6 +216,9 @@ func (s *Server) handleAppProbe(w http.ResponseWriter, req *http.Request) {
 		appReq.Header[header.Name] = []string{header.Value}
 	}
 
+	// Pass the user agent from the original request
+	appReq.Header["User-Agent"] = []string{req.UserAgent()}
```

Comment:
Isn't this covered by L216?

Reply:
No, L216 only covers headers defined in the [`httpHeaders`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#httpgetaction-v1-core) section of the probe. It's not those coming from the upstream request.

Reply:
The probe request only comes from kubelet?  

Do you really care about the `"User-Agent"`?

Reply:
I know of at least one app (Knative Serving Activator) that does depend on it (they've will be adding a header now as well) for the health check.

I'm not going to argue for/against using the user-agent for health checking, but I would argue that from an application perspective, the Istio probe rewriting shouldn't result in a different request arriving. It would be worthwhile probably also forwarding all of the headers, but the User-Agent + the ones specified in the probe are probably sufficient for an application.

Reply:
If additional headers may be added, the i suggest propagate all req.Headers, not only `user-agent`. WDYT?

Reply:
That's probably a better option, yeah. I'll update PR shortly.

Reply:
I have updated the to forward all incoming headers.